x-pack/plugins/apm/common/utils/get_preferred_bucket_size_and_data_source.test.ts

```diff
@@ -4,48 +4,55 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { ApmDataSource } from '../data_source';
+import { ApmDataSourceWithSummary } from '../data_source';
 import { ApmDocumentType } from '../document_type';
 import { RollupInterval } from '../rollup';
 import {
   getPreferredBucketSizeAndDataSource,
   intervalToSeconds,
 } from './get_preferred_bucket_size_and_data_source';
 
-const serviceTransactionMetricSources: ApmDataSource[] = [
+const serviceTransactionMetricSources: ApmDataSourceWithSummary[] = [
   {
     documentType: ApmDocumentType.ServiceTransactionMetric,
     rollupInterval: RollupInterval.OneMinute,
+    isSummaryFieldAvailable: true,
   },
   {
     documentType: ApmDocumentType.ServiceTransactionMetric,
     rollupInterval: RollupInterval.TenMinutes,
+    isSummaryFieldAvailable: true,
   },
   {
     documentType: ApmDocumentType.ServiceTransactionMetric,
     rollupInterval: RollupInterval.SixtyMinutes,
+    isSummaryFieldAvailable: true,
   },
 ];
 
-const txMetricSources: ApmDataSource[] = [
+const txMetricSources: ApmDataSourceWithSummary[] = [
   {
     documentType: ApmDocumentType.TransactionMetric,
     rollupInterval: RollupInterval.OneMinute,
+    isSummaryFieldAvailable: true,
```

Comment:
Should we call out that this is the _duration_ summary field?

```suggestion
    hasDurationSummaryField: true,
```